eZ/Publish/Core/Persistence/Legacy/Content/Location/Trash/Handler.php

```diff
@@ -197,37 +200,51 @@ public function findTrashItems(Criterion $criterion = null, $offset = 0, $limit
      */
     public function emptyTrash()
     {
+        $resultList = new TrashedItemDeleteResultList();
         $trashedItems = $this->findTrashItems();
         foreach ($trashedItems as $item) {
-            $this->delete($item);
+            $resultList->items[] = $this->delete($item);
         }
 
         $this->locationGateway->cleanupTrash();
+
+        return $resultList;
     }
 
     /**
      * Removes a trashed location identified by $trashedLocationId from trash
      * Associated content has to be deleted.
      *
      * @param int $trashedId
+     *
+     * @return \eZ\Publish\SPI\Persistence\Content\Location\TrashedItemDeleteResult
```

Comment:
Typo in FQCN:
```suggestion
     * @return \eZ\Publish\SPI\Persistence\Content\Location\Trash\TrashedItemDeleteResult
```
**but I'd go for API as said before.**